src/libs/actions/Task.js

```diff
@@ -0,0 +1,139 @@
+import Onyx from 'react-native-onyx';
+import CONST from '../../CONST';
+import ONYXKEYS from '../../ONYXKEYS';
+import * as API from '../API';
+import * as ReportUtils from '../ReportUtils';
+import * as Report from './Report';
+import Navigation from '../Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+
+/**
+ * Clears out the task info from the store
+ */
+function clearOutTaskInfo() {
+    Onyx.set(ONYXKEYS.TASK, null);
+}
+
+/**
+ * Assign a task to a user
+ * Function title is createTask for consistency with the rest of the actions
+ * and also because we can create a task without assigning it to anyone
+ * @param {String} parentReportID
+ * @param {String} title
+ * @param {String} description
+ * @param {String} assignee
+ *
+ */
+
+function createTaskAndNavigate(currentUserEmail, parentReportID, title, description, assignee) {
+    // Grab the assigneeChatReportID if there is an assignee
+    const assigneeChatReportID = ReportUtils.getChatByParticipants([assignee]).reportID;
+
+    // Optimistically created Action on the parent
+    const optimisticCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(parentReportID);
```

Comment:
Still don't need this a created reportAction on the parent (chat report) so can be removed. It should already exist and therefore already have a created reportAction